typescript/src/proof.ts

```diff
@@ -70,3 +79,253 @@ function createMerkleProof(txMerkleBranch: TransactionMerkleBranch): string {
+      current = computeHash256(
+        intermediateNodesHashes.slice(i, i + 64) + current
+      ).toString()
+    } else {
+      current = computeHash256(
+        current + intermediateNodesHashes.slice(i, i + 64)
+      ).toString()
+    }
+    idx = idx >> 1
+  }
+
+  if (current !== merkleRootHash) {
+    throw new Error(
+      "Transaction Merkle proof is not valid for provided header and transaction hash"
+    )
+  }
+}
+
+/**
+ * Validates a chain of consecutive block headers by checking each header's
+ * difficulty, hash, and continuity with the previous header. This function can
+ * be used to validate a series of Bitcoin block headers for their validity.
+ * @param blockHeaders An array of block headers that form the chain to be
+ *        validated.
+ * @param previousEpochDifficulty The difficulty of the previous Bitcoin epoch.
+ * @param currentEpochDifficulty The difficulty of the current Bitcoin epoch.
+ * @dev The block headers must come from Bitcoin epochs with difficulties marked
+ *      by the previous and current difficulties. If a Bitcoin difficulty relay
+ *      is used to provide these values and the relay is up-to-date, only the
+ *      recent block headers will pass validation. Block headers older than the
+ *      current and previous Bitcoin epochs will fail.
+ * @throws {Error} If any of the block headers are invalid, or if the block
+ *         header chain is not continuous.
+ * @returns An empty return value.
+ */
+function validateBlockHeadersChain(
+  blockHeaders: BlockHeader[],
+  previousEpochDifficulty: BigNumber,
+  currentEpochDifficulty: BigNumber
+) {
+  let requireCurrentDifficulty: boolean = false
+  let previousBlockHeaderHash: Hex = Hex.from("00")
+
+  for (let index = 0; index < blockHeaders.length; index++) {
+    const currentHeader = blockHeaders[index]
+
+    // Check if the current block header stores the hash of the previously
+    // processed block header. Skip the check for the first header.
+    if (index !== 0) {
+      if (
+        !previousBlockHeaderHash.equals(currentHeader.previousBlockHeaderHash)
+      ) {
+        throw new Error("Invalid headers chain")
+      }
+    }
+
+    const difficultyTarget = bitsToDifficultyTarget(currentHeader.bits)
+
+    const currentBlockHeaderHash = computeHash256(
+      serializeBlockHeader(currentHeader)
+    )
+
+    // Ensure the header has sufficient work.
+    if (
+      hashLEToBigNumber(currentBlockHeaderHash.toString()).gt(difficultyTarget)
+    ) {
+      throw new Error("Insufficient work in the header")
+    }
+
+    // Save the current block header hash to compare it with the next block
+    // header's previous block header hash.
+    previousBlockHeaderHash = currentBlockHeaderHash
+
+    // Check if the stored block difficulty is equal to previous or current
+    // difficulties.
+    const difficulty = targetToDifficulty(difficultyTarget)
+
+    if (previousEpochDifficulty.eq(1) && currentEpochDifficulty.eq(1)) {
```

Comment:
This is a special case for Bitcoin Testnet.
When both difficulties are set to `1` we skip difficulties check.
I assumed the relay used in our testnet environment always returns `1` for current and previous difficulties.

We could also temporarily pass `1` as previous and current difficulties for Mainnet, unit we have the `LightRelay` up and running.